Feature/remake

AndreKunitz opened the pull request:
Realizada as mudanças que discutimos na reunião.

AndreKunitz committed:
Refactor packaging

This commit will adjust packaging and package names according to REST
architecture best practices.

Also, the application properties file extension was change for better
readability.

AndreKunitz committed:
Polish entities modeling

Added new properties both in Log and User entities with validations
constraints.

Rework of database to deal with the new modeling.

AndreKunitz committed:
Register new user

This commit implement register method in user resource for registering
new users.

To not expose model propeties in resquest body, it was used
UserRegistrationDTO for transmitting only the necessary fields.

To map UserRegistrationDTO to User model, it was used MapStruct plugin.

AndreKunitz committed:
Polish user registration

Cleaned user mapper class to deal with both user request DTOs and user
response DTOs.

Added swaggeer annotations to better document the API.

Register method in user endpoint now returns a body with proper info.

AndreKunitz committed:
Rework packaging

This commit will change packaging and package names according to
traditional REST applications.

Database name was also changed for the following up application name
change.

AndreKunitz committed:
Change application name

AndreKunitz committed:
Add an abstract audit and log aggregate classes

The createdAt and modifiedAt auditing properties were moved to an
abstract class.

Removed Active and Deleted enum from Status, since now logs will only
be marked as Archived.

Added a sketch model for the log aggregate.

Changed timestamp property in StandardErro to use java.time.

eduardodecarvalho commented on src/main/java/br/com/codenation/softlog/service/UserService.java:
```diff
@@ -0,0 +1,34 @@
+    private UserRepository userRepository;
+
+    @Autowired
+    private UserMapper mapper;
```
Não colocamos @Autowired em entidades.

eduardodecarvalho commented on src/main/java/br/com/codenation/softlog/service/UserService.java:
```diff
@@ -0,0 +1,34 @@
+//        TODO:    - Revisar regra sobre Token
+//                 - Nomes diferentes para cada map? (toUser() e toUserResponseDTO)
+
+        user.setToken(UUID.randomUUID().toString());
```
setToken() não está funcionando. Devemos comentar essa linha, visto que ainda não temos nenhuma definição a respeito do token.

eduardodecarvalho commented on src/main/java/br/com/codenation/softlog/model/User.java:
```diff
@@ -0,0 +1,44 @@
+    private String password;
+
+    @NotBlank
+    private String token;
```
Se o token será imutável não podemos habilitar o setToken()

AndreKunitz committed:
Fix dependency injection and make token immutable

Added API documentation with swagger.

Fixed DI using lombok in UserService class.

Made token immutable according to business logic.

AndreKunitz commented on src/main/java/br/com/codenation/softlog/dto/response/UserResponseDTO.java:
```diff
@@ -0,0 +1,23 @@
+
+    // TODO:
+    //      - Retornar id ou token?
+
```
Aqui vamos retornar o ID do banco ou o TOKEN? naõ vejo razão para o usuário saber o ID do banco...

AndreKunitz commented on src/main/java/br/com/codenation/softlog/model/User.java:
```diff
@@ -0,0 +1,47 @@
+    @Setter(AccessLevel.NONE)
+    @NotBlank
+    private String token = UUID.randomUUID().toString();
+
```
Alguém da uma mão aqui... Não consegui fazer a atribuição do token no construtor usando o lombok.

AndreKunitz commented on src/main/java/br/com/codenation/softlog/service/UserService.java:
```diff
@@ -0,0 +1,32 @@
+
+    private UserRepository userRepository;
+    private UserMapper mapper;
+
```
Aqui usei o @AllArgsConstructor. Achei mais elegante do que fazer @Autowired UserRepo e new UserMapper().

AndreKunitz committed:
Adjustmentes for class

DouglasKlafkeScheibler committed:
Refatora entidade Log e User e ajusta migrações

aguilardias committed:
test log aggregate

aguilardias committed:
Merge branch 'feature/remake' of git@github.com:codenation-dev/squad-1-ad-java-softplan-2.git into feature/remake
